server/src/main/java/com/unosquare/carmigo/controller/UserController.java

```diff
@@ -0,0 +1,125 @@
+        createAuthenticationViewModel, CreateAuthenticationDTO.class);
+    final GrabAuthenticationDTO grabAuthenticationDTO = userService.createAuthenticationToken(createAuthenticationDTO);
+    final AuthenticationViewModel authenticationViewModel = modelMapper.map(
+        grabAuthenticationDTO, AuthenticationViewModel.class);
+    return new ResponseEntity<>(authenticationViewModel, HttpStatus.CREATED);
+  }
+
+  @GetMapping(value = "/profile", produces = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<PlatformUserViewModel> getPlatformUserProfile() {
+    return ResponseEntity.ok(userControllerHelper.getPlatformUserById(0));
+  }
+
+  @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.CREATED)
+  public ResponseEntity<PlatformUserViewModel> createPlatformUser(
+      @Valid @RequestBody final CreatePlatformUserViewModel createPlatformUserViewModel) {
+    final CreatePlatformUserDTO createPlatformUserDTO = modelMapper.map(
+        createPlatformUserViewModel, CreatePlatformUserDTO.class);
+    final GrabPlatformUserDTO grabPlatformUserDTO = userService.createPlatformUser(createPlatformUserDTO);
+    final PlatformUserViewModel platformUserViewModel = modelMapper.map(
+        grabPlatformUserDTO, PlatformUserViewModel.class);
+    return new ResponseEntity<>(platformUserViewModel, HttpStatus.CREATED);
+  }
+
+  @PatchMapping(consumes = "application/json-patch+json")
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public ResponseEntity<PlatformUserViewModel> patchPlatformUser(@RequestBody final JsonPatch patch) {
+    return new ResponseEntity<>(userControllerHelper.patchPlatformUserById(0, patch), HttpStatus.ACCEPTED);
+  }
+
+  @DeleteMapping
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public ResponseEntity<?> deletePlatformUser() {
+    userControllerHelper.deletePlatformUserById(0);
+    return ResponseEntity.noContent().build();
+  }
+
+  @GetMapping(value = "/drivers/profile", produces = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<DriverViewModel> getDriverProfile() {
+    return ResponseEntity.ok(userControllerHelper.getDriverById(0));
+  }
+
+  @PostMapping(value = "/drivers", produces = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.CREATED)
+  public ResponseEntity<DriverViewModel> createDriver(
+      @Valid @RequestBody final CreateDriverViewModel createDriverViewModal) {
+    final DriverViewModel driverViewModel = userControllerHelper.createDriverById(0, createDriverViewModal);
+    return new ResponseEntity<>(driverViewModel, HttpStatus.CREATED);
+  }
+
+  @DeleteMapping(value = "/drivers")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public ResponseEntity<?> deleteDriver() {
+    userControllerHelper.deleteDriverById(0);
+    return ResponseEntity.noContent().build();
+  }
+
+  @GetMapping(value = "/passengers/profile", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
AGain, I am not sure the value of this endpoint.

Reply:
This is to get information of the current passenger.